ts/components/conversation/right-panel/overlay/disappearing-messages/TimeOptions.tsx

```diff
@@ -0,0 +1,45 @@
+import { isEmpty } from 'lodash';
+import React from 'react';
+import { TimerOptionsArray } from '../../../../../state/ducks/timerOptions';
+import { PanelButtonGroup, PanelLabel } from '../../../../buttons/PanelButton';
+import { PanelRadioButton } from '../../../../buttons/PanelRadioButton';
+
+type TimerOptionsProps = {
+  options: TimerOptionsArray | null;
+  selected: number;
+  setSelected: (value: number) => void;
+  hasOnlyOneMode?: boolean;
+  disabled?: boolean;
+};
+
+export const TimeOptions = (props: TimerOptionsProps) => {
+  const { options, selected, setSelected, hasOnlyOneMode, disabled } = props;
+
+  if (!options || isEmpty(options)) {
+    return null;
+  }
+
+  return (
+    <>
+      {!hasOnlyOneMode && <PanelLabel>{window.i18n('timer')}</PanelLabel>}
+      <PanelButtonGroup>
+        {options.map(option => {
+          return (
+            <PanelRadioButton
+              key={option.name}
+              text={option.name}
+              value={option.name}
+              isSelected={selected === option.value}
+              onSelect={() => {
+                setSelected(option.value);
+              }}
+              noBackgroundColor={true}
+              disabled={disabled}
+              dataTestId={`disappear-time-${option.name.replace(' ', '-')}-option`}
```

Comment:
I've seen if there is a way to generate an english version but we would need to add some functions in i18n so instead of chose to use the seconds value as the unique id. Then it can be used without worry about localisation.